### ISSUE TITLE
Training doesn't work

### ISSUE THREAD

**Issue by ilanshib** [enhuiz/vall-e#57]
Hi,

I tried following the Vall-E demo flow at https://blog.paperspace.com/training-vall-e-from-scratch-on-your-own-voice-samples/ and sample the model to generate an audio file with my own voice but the result was noise. I noticed and changed the following:

1. In the original code max_phones in config.py was set to 50. It filtered out all the training files but one so I changed it to 250.
2. I ran a few training rounds and played with config/libri/ar.yml and config/libri/nar.ymel values. For example - I extended max_iter to 2048 and changed the batch_size to 4 or 8 or 16 and even 32 but the model.loss remained quite high (~3 for ar and ~5 for nar).

Did anyone manage to get decent results and if so, did you apply any changes to the original paperspace demo project provided at https://console.paperspace.com/github/gradient-ai/vall-e?machine=Free-GPU?

Thanks!
Ilan

**Comment by agupta54** [comment]
I tried to train on LibriTTS data. For me the ar loss converged to ~1.4 for and ~3.5 for nar after 60,000 steps and a batch size of 64. The hyp and ref in the logs folder actually become sound quite close after training. But when I am exporting the checkpoints, I am not able to reproduce meaningful audios from prompts.

**Comment by yiwei0730** [comment]
Yes, I trained on the LibriTTS-360hr data and in the quarter model. 
It didn't work. Maybe need to try more bigger data or more bigger model parameters.

**Comment by huukim136** [comment]
I trained LibriTTS with the quarter setting,  got ar loss ~2.5 and nar loss ~3.2.
When inference, I got something like this
[valle.zip](https://github.com/enhuiz/vall-e/files/10878064/valle.zip)
But I noticed there are many issues with my current model such as word skipping, word repetition, timbre changing, ending prematurely.
I'm not sure if I have hit the maximum capacity of the quarter model or not

**Comment by yiwei0730** [comment]
@huukim136  how many data hours you used, my ar loss ~ 1.5  and nar loss ~ 3.6，but get the difference inference result with you.
[valle.zip](https://github.com/enhuiz/vall-e/files/10878175/valle.zip)
Here is my config of ar and  nar model
data_dirs: [data/LibriTTS/]
spkr_name_getter: "lambda p: p.parts[-3]"
model: ar-quarter
batch_size: 48
eval_batch_size: 48
eval_every: 10_000

data_dirs: [data/LibriTTS/]
spkr_name_getter: "lambda p: p.parts[-3]"
model: nar-quarter
batch_size: 48
eval_batch_size: 48

**Comment by huukim136** [comment]
@yiwei0730 I used around ~150 hours for training. 
batch size: 16.

**Comment by agupta54** [comment]
@huukim136 are these audios from the log folder or generated after exporting the checkpoints? When I export checkpoints my sound quality is more like @yiwei0730 
my configs are: 
```
data_dirs: [data/LibriTTS/]
spkr_name_getter: "lambda p: p.parts[-3]"

model: ar
batch_size: 64
eval_batch_size: 24
eval_every: 10_000
max_iter: 60000
sampling_temperature: 1.0
```
```
data_dirs: [data/LibriTTS/]
spkr_name_getter: "lambda p: p.parts[-3]"

model: nar
batch_size: 64
eval_batch_size: 24
eval_every: 1_000
max_iter: 60000
sampling_temperature: 0.2
```

**Comment by huukim136** [comment]
wait @yiwei0730, how did you get that low ar loss (1.5)?
Is it eval loss of training loss? How many steps it took to get that?
Can you share your learning rate and training schedule also?

**Comment by huukim136** [comment]
@agupta54 they are generated after exporting checkpoints. Everything was unseen for the model.

**Comment by yiwei0730** [comment]
First the config problem, you can set some in config dataset! Which i gave is the setting in config(Like the agupta54 setting), but other setting is make in the progress.
AR trainer log in 1000000
 vall_e.utils.trainer - INFO - 
{
  "model.loss": 1.53125,
  "model.lr": 1.1991991991991991e-06,
  "model.grad_norm": 0.7619398832321167,
  "model.elapsed_time": 0.3643155097961426,
  "model.engine_step": 1000000,
  "model.loss.nll": 1.53125,
  "elapsed_time": 0.3643155097961426,
  "wall_time": 1676929853.196073,
  "global_step": 1000000
}
NAR trainer log in 1000000step
vall_e.utils.trainer - INFO - 
{
  "model.loss": 3.634765625,
  "model.lr": 1.1991991991991991e-06,
  "model.grad_norm": 1.1556732654571533,
  "model.elapsed_time": 0.3783080577850342,
  "model.engine_step": 1000000,
  "model.loss.nll": 3.634765625,
  "elapsed_time": 0.3783080577850342,
  "wall_time": 1677346919.0852554,
  "global_step": 1000000
}

**Comment by yiwei0730** [comment]
@huukim136  Can you show the setting in your log.txt ?

**Comment by huukim136** [comment]
@yiwei0730 I'm not sure what log.txt file you are refering to but my training config is like this:
![Screen Shot 2023-03-03 at 2 29 57 PM](https://user-images.githubusercontent.com/22656013/222639072-b05e8759-1cf6-4d98-a08c-c7f140fbe9f7.png)
actually i was asking you about  the values for this config

**Comment by huukim136** [comment]
@yiwei0730 your ar loss 1.5 is evaluation loss or training loss?

**Comment by yiwei0730** [comment]
![image](https://user-images.githubusercontent.com/62238248/222639995-cf806d0c-7dfd-44b6-ac86-1d8548f57e12.png)
I see. This is the cfg.json. 
My loss is in the log.txt which is in the logs dir and stay together with the cfg.json. 
![image](https://user-images.githubusercontent.com/62238248/222641094-ab05f097-39d4-439a-ac7b-2582813bb538.png)
I just saw one loss in one step I think it is the training loss
![image](https://user-images.githubusercontent.com/62238248/222640449-a01d4190-7897-46d4-8dd2-615250cb68b7.png)

**Comment by yiwei0730** [comment]
@huukim136 There is some setting different between us.
But I just fork the code in recent....。

**Comment by yiwei0730** [comment]
Oh i see, your config setting is fitting the thesis setting, but the config in the github code is not
@huukim136  Did you just rewrite the code? example the config setting and the optimizer for using AdamW

**Comment by agupta54** [comment]
> @agupta54 they are generated after exporting checkpoints. Everything was unseen for the model.

For that the quality is really good. I was using the default configs from the repo apart from number of steps and batch size. Can you share yours?

**Comment by huukim136** [comment]
apart from the configs i almost kept the code intact. I already shared my configs. check the above comment

**Comment by huukim136** [comment]
@agupta54 it's not good at all. those are the samples that I cherry-picked. any TTS model can easily pass the quality of the model I have now tbh

**Comment by agupta54** [comment]
> @agupta54 it's not good at all. those are the samples that I cherry-picked. any TTS model can easily pass the quality of the model I have now tbh

got it!

**Comment by ilanshib** [comment]
I am using a 100hrs database (checkout https://www.openslr.org/12/ it has a few DBs there) and up till now executed about 10,000 ephochs. loss is ~3 for AR and sampling results are only noise. Tried to increase the Learning Rate (min_lr and max_lr) by a factor of 10 but it seemed to make things worse so now I tried to decrease it by a factor of 5 and I'm waiting to see see if it makes a difference (I simply edit vall-e/vall_e/config.py and apply any change I want).

Can anyone share the model you trained so we can compare the outputs?

**Comment by yiwei0730** [comment]
our ouputs is on the top in valle.zip, you can hear the voice, maybe should took the good setting and use the normal size paramters, by the way . I think the data needs more

**Comment by yiwei0730** [comment]
@huukim136 So, you change some setting to be similar to the thesis, ex: optimizer, warmup step , warmup learning rate.
Can i know other setting in your code changing? Can i have your cfs.json ?

**Comment by huukim136** [comment]
@yiwei0730 the above figure is pretty much everything of my config file

**Comment by ilanshib** [comment]
@yiwei0730, the zip files you sent are much much better than anything I got so far.  My question was whether anyone can post a zip with the resulting trained model so we can use it for sampling and comparing results. 

To my understanding it is stored in subdirectories 'ckpts' and maybe 'zoo', located under vall-e. (they both include .pt files).

This could really help understand where I am along with the loss value.

**Comment by cantabile-kwok** [comment]
I trained on LibriTTS 460hrs data. The NAR model's loss exploded after 10 days of training, and there is no previous model checkpoints so I lost the best model so far. This is a depressing result and I wonder if anyone has the similar experiences.
